docs/ha-considerations.md

```diff
@@ -0,0 +1,337 @@
+# High Availability Considerations
+
+## Overview
+
+When setting up a productive cluster, high availability, i.e. the cluster's ability to remain operational even if some control plane or worker nodes fail, is usually a requirement. For worker nodes, assuming that there are enough of them, this is part of the very cluster functionality; however redundancy of control plane nodes and `etcd` instances needs to be catered for when planning and setting up a cluster.
+
+`kubeadm` supports setting up of multi control plane and multi `etcd` clusters (see [Creating Highly Available clusters with kubeadm](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/high-availability/), part of the official Kubernetes documentation, for a step-by-step guide). However there are some more things to consider and set up which are not part of Kubernetes itself and hence not covered there. This document provides some additional information and examples useful when planning and bootstrapping HA clusters with `kubeadm`.
```

Comment:
i'd remove
`, part of the official Kubernetes documentation,`

---

`However there are some more things` -> `However there are some aspects...`

---

`not covered there.` -> `not covered in the project documentation.`

---